builder/image/templates/builder

```diff
@@ -56,7 +56,12 @@ mkdir -p $BUILD_DIR $CACHE_DIR
 TMP_DIR=$(mktemp -d --tmpdir=$BUILD_DIR)
 
 cd $REPO_DIR
-git archive $GIT_SHA | tar -xmC $TMP_DIR
+git clone $REPO_DIR/ "$TMP_DIR/"
+pushd "$TMP_DIR"
+export GIT_DIR="$TMP_DIR/.git" GIT_WORK_TREE="$TMP_DIR"
+git config core.bare false
+git reset --hard $GIT_SHA
```

Comment:
why not a `git checkout` instead? That way we're not destroying the tree's history